appcues/src/main/java/com/appcues/ui/ExperienceRenderer.kt

```diff
@@ -103,7 +106,8 @@ internal class ExperienceRenderer(
         experiment?.let { analyticsTracker.track(it) }
 
         when (stateMachine.handleAction(StartExperience(this))) {
-            is Success -> true
+            // removes possible preview experience from hashMap if it was showed properly
+            is Success -> true.also { previewExperiences.remove(renderContext) }
```

Comment:
it looks like on the iOS side that the preview gets [removed on completion](https://github.com/appcues/appcues-ios-sdk/pull/441/files#diff-e9daefd60bacc377827cb6a2002bcd0fe84cd5b3ec407cd9ab2a6a5dd676a37fR262). That would correspond to L177 the completion handler of the lifecycle tracker in this class, I think.

granted, completion may not happen for embeds very often, right? @mmaatttt makes me wonder if removal here on successful start is better?

Reply:
I guess my idea is that whenever it StartExperience with success we can basically clean up the preview dictionary right, that would work ok for all

Reply:
hey @mmaatttt just checking on this one - I think I'd agree - removing a preview as soon as it successfully renders seems appropriate.